libimage/layer_tree.go

```diff
@@ -91,17 +91,27 @@ func (l *layerNode) repoTags() ([]string, error) {
 
 // layerTree extracts a layerTree from the layers in the local storage and
 // relates them to the specified images.
-func (r *Runtime) layerTree(ctx context.Context, images []*Image) (*layerTree, error) {
-	layers, err := r.store.Layers()
-	if err != nil {
-		return nil, err
-	}
-
-	if images == nil {
-		images, err = r.ListImages(ctx, nil, nil)
+// layerTree require both parameters not empty or nil, to avoid inconsistences
+// between layers and images. If both parameters are nil then layers and images will be obtained from local storage.
+// You can obtain consistent layer and images with storage.MultiList.
+func (r *Runtime) layerTree(images []*Image, layers []storage.Layer) (*layerTree, error) {
+	switch is_consistent := (images != nil || layers == nil) && (images == nil || layers != nil); is_consistent {
+	case !is_consistent: // inconsistent data
```

Comment:
This is mixing the “switch on a value” and “switch with no value, each `case` is a condition” modes, and I don’t think that works. E.g. I read this case to be `case is_consistent == !is_consistent`